lib/SIL/SILConstants.cpp

```diff
@@ -479,20 +496,27 @@ void SymbolicValue::emitUnknownDiagnosticNotes(SILLocation fallbackLoc) {
     break;
   }
 
-  auto &module = badInst->getModule();
+  emitNoteDiagnostic(badInst, unknown.second, fallbackLoc, error);
 
-  auto loc = skipInternalLocations(badInst->getDebugLocation()).getLocation();
-  if (loc.isNull()) {
-    // If we have important clarifying information, make sure to emit it.
-    if (unknown.second == UnknownReason::Default ||
-        fallbackLoc.isNull())
-      return;
-    loc = fallbackLoc;
+  auto &module = badInst->getModule();
+  auto &SM = module.getASTContext().SourceMgr;
+  bool emittedFirstNote = false;
+  unsigned originalDiagnosticLineNumber =
+      SM.getLineNumber(fallbackLoc.getSourceLoc());
+  for (auto &sourceLoc : llvm::reverse(getUnknownCallStack())) {
+    // Skip known sources.
+    if (!sourceLoc.isValid())
+      continue;
+    // Also skip notes that point to the same line as the original error, for
+    // example in:
+    //   #assert(foo(bar()))
+    // it is not useful to get three diagnostics referring to the same line.
+    if (SM.getLineNumber(sourceLoc) == originalDiagnosticLineNumber)
+      continue;
+
+    auto diag = emittedFirstNote ? diag::constexpr_called_from
```

Comment:
also consider adding a test case where diag::constexpr_called_from is exercised 2 or more times?